src/core/public/application/application_service.tsx

```diff
@@ -115,7 +115,20 @@ export class ApplicationService {
     context,
     http: { basePath },
     redirectTo = (path: string) => {
-      window.location.assign(path);
+      try {
+        const location = new URL(path, window.location.origin);
+
+        if (location?.origin === window.location.origin) {
+          window.location.assign(path);
+        } else {
+          window.location.assign(
+            basePath.prepend(`/internal/external_redirect/goto?next=${encodeURIComponent(path)}`)
+          );
```

Comment:
- Where is this route defined? Couldn't find it in the PR changes, did I miss it?

- I see we have a `ExternalUrlService` client-side service, why using a redirection instead of performing the validation directly on the client-side?

- This doesn't really protect against 'plain' links (`<a>` tags that could be displayed into markdown for example), and will force teams to manually implement the check via the API. However, I guess this is kinda the best we can do without some ugly top-level global handler that would probably causes the same issue we encountered when trying to implement it for the cross-app link navigation.

Reply:
> Where is this route defined? Couldn't find it in the PR changes, did I miss it?

Ah, this was a holdover from a previous implementation. There should be no changes to this file.

> This doesn't really protect against 'plain' links (<a> tags that could be displayed into markdown for example), and will force teams to manually implement the check via the API (when not using navigateToUrl at least). However, I guess this is kinda the best we can do without some ugly top-level global handler that would probably causes the same issue we encountered when trying to implement it for the cross-app link navigation.

That's true, and my intent was not to add any sort of protection in this PR. My goal was to provide the service that other teams could consume, in whatever way makes the most sense for them. Solving all of these use cases would blow up the scope of this work, and isn't something we've honestly allocated time for.

Reply:
> Ah, this was a holdover from a previous implementation. There should be no changes to this file.

Ok, so the PR is supposed to just expose this service from `core`, without wiring it automatically anywhere, right? In that case, it sounds good to me, the implementation looks fine.

Reply:
> Ok, so the PR is supposed to just expose this service from core, without wiring it automatically anywhere, right?

Yup, that's exactly right!